### PR TITLE
ci: Fix unittests of logits processor

### PR DESCRIPTION
**Pull request by yzh119**
## 📌 Description

There are 8 tests failed in test_logits_processor, the bug was introduced in #1170 and because of the minor discrepancy between torch softmax and the flashinfer softmax output.

This PR changes the default softmax implementation to torch's, and only enables flashinfer softmax when fusing with temperature.

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

### CODE REVIEW

**Review by yongwww — APPROVED** [review]

**Comment by bkryu** [comment]
Confirm that the failures in test_logits_processor  are fixed with this PR
